src/internal_modules/roc_core/ipool.h

```diff
@@ -54,8 +54,7 @@ class IPool {
 //! Placement new for core::IPool.
 //! @note
 //!  nothrow forces compiler to check for NULL return value before calling ctor.
-inline void* operator new(size_t size, roc::core::IPool& pool) throw() {
-    roc_panic_if_not(size <= pool.object_size());
```

Comment:
@gavv 
Note: I had to remove this only good use of object_size() as well.

Reply:
I think we can add `size` parameter to allocate(). Implementation can then check that requested size fits within slot size. This way we can be sure that any allocation on pool is safe (no matter via new or directly via allocate).